### PR TITLE
Guard for mentions suggestions list clipped by bottom edge of window

### PR DESCRIPTION
**Pull request by coxom**
When the mentions popup is near the bottom of the screen it will be moved up so it doesn't get cut.
This correction will only be applied when the property suggestionsPortalHost is not being used.

### CODE REVIEW

File: src/MentionsInput.js [hunk 1]
```diff
@@ -530,7 +530,19 @@ class MentionsInput extends React.Component {
       } else {
         position.left = left
       }
-      position.top = caretPosition.top - highlighter.scrollTop
+
+      // guard for mentions suggestions list clipped by bottom edge of window
+      const tooltipBottom =
+        this.containerRef.getBoundingClientRect().top +
```
Comment (frontendphil) [inline src/MentionsInput.js]: Since `refs` can be `null` it seems like you need to add an extra check here in case `containerRef` isn't set yet.

Comment (coxom) [reply src/MentionsInput.js]: Added the check to both left and top

**Comment by DJTB** [comment]
Anything I can do to help this through?

**Comment by nummi** [comment]
I would love to have this feature as well. Thank you for implementing it.

**Comment by adityatandon007** [comment]
When will this be merged. Needed this badly in one of my projects. Please add this feature and push it in the next release.

**Comment by BoyWithSilverWings** [comment]
Is there any update on this? The PR looks ready.

**Comment by jfschwarz** [comment]
The proposed implementation does not look like a robust solution to the problem. While it guards against clipping at the bottom edge it will instead cause clipping at the top edge for long suggestions lists.

There are some other undesired behaviors. For example, when looking at the "Advanced options" example on our demo page, it opens the suggestions overlay to the top even though there would be enough space to open it to the bottom.

If you want to help get this PR through, please add examples to the demo page and propose a more complete solution.

**Pull request closed without merging**